app/Services/CampaignService.php

```diff
@@ -70,6 +70,7 @@ public function getPostTotals($campaign)
                     DB::raw('SUM(case when posts.status = "rejected" && posts.deleted_at is null then 1 else 0 end) as rejected_count'))
                 ->where('signups.campaign_id', '=', $campaign['id'])
                 ->groupBy('signups.campaign_id')
+                ->whereNull('signups.deleted_at')
```

Comment:
Do we have an index on this column? If not, that might be why this is timing out for you.